experimental/sdk/webpubsub-socketio-extension/src/SIO/index.ts

```diff
@@ -49,4 +76,33 @@ export async function useAzureSocketIO(
   return this;
 }
 
+function getDefaultNegotiateHandler(
+  engine: WebPubSubEioServer
+): (
+  req: IncomingMessage,
+  res: ServerResponse,
+  getClientAccessTokenUrl: (options?: GenerateClientTokenOptions) => Promise<string>
+) => Promise<void> {
+  return async (
+    req: IncomingMessage,
+    res: ServerResponse,
+    getClientAccessTokenUrl: (options?: GenerateClientTokenOptions) => Promise<string>
+  ): Promise<void> => {
+    let statusCode = 400,
+      message = "Bad Request";
+    try {
+      const username = parse(req.url || "", true).query["username"] as string;
+      const endpointWithToken = await getClientAccessTokenUrl({ userId: username ?? "" });
```

Comment:
the feature this default handler supports is quite limited, can user pass in roles or initial groups?

Reply:
Now it supports both userId and expiration time